lib/rules/prefer-template.js

```diff
@@ -38,6 +38,45 @@ function getTopConcatBinaryExpression(node) {
     return currentNode;
 }
 
+/**
+ * Determines whether a given node is a octal escape sequence
+ * @param {ASTNode} node A node to check
+ * @returns {boolean} `true` if the node is an octal escape sequence
+ */
+function isOctalEscapeSequence(node) {
+
+    // No need to check TemplateLiterals – would throw error with octal escape
+    const isStringLiteral = node.type === "Literal" && typeof node.value === "string";
+
+    if (!isStringLiteral) {
+        return false;
+    }
+
+    const match = node.raw.match(/^([^\\]|\\[^0-7])*\\([0-3][0-7]{1,2}|[4-7][0-7]|[0-7])/);
```

Comment:
I think the second part of this regular expression could be simplified from this:

```
([0-3][0-7]{1,2}|[4-7][0-7]|[0-7])
```

to this:

```
([0-7]{1,3})
```

...since it seems like the `[0-7]` character class will always match something even if the other cases don't.